InterfaceTPL/tplUnitTests.cpp

```diff
@@ -0,0 +1,139 @@
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * UT-BATTELLE, LLC AND THE UNITED STATES GOVERNMENT MAKE NO REPRESENTATIONS AND DISCLAIM ALL WARRANTIES, BOTH EXPRESSED AND IMPLIED.
+ * THERE ARE NO EXPRESS OR IMPLIED WARRANTIES OF MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE, OR THAT THE USE OF THE SOFTWARE WILL NOT INFRINGE ANY PATENT,
+ * COPYRIGHT, TRADEMARK, OR OTHER PROPRIETARY RIGHTS, OR THAT THE SOFTWARE WILL ACCOMPLISH THE INTENDED RESULTS OR THAT THE SOFTWARE OR ITS USE WILL NOT RESULT IN INJURY OR DAMAGE.
+ * THE USER ASSUMES RESPONSIBILITY FOR ALL LIABILITIES, PENALTIES, FINES, CLAIMS, CAUSES OF ACTION, AND COSTS AND EXPENSES, CAUSED BY, RESULTING FROM OR ARISING OUT OF,
+ * IN WHOLE OR IN PART THE USE, STORAGE OR DISPOSAL OF THE SOFTWARE.
+ */
+
+#ifndef __TPL_UNIT_TESTS_CPP
+#define __TPL_UNIT_TESTS_CPP
+
+#include "tplUnitTests.hpp"
+#include "tsgBlasWrappers.hpp"
+#include <string>
+#include <math.h>
+#include <cmath>
+
+bool testTridiagonal();
+
+TplUnitTester::TplUnitTester() : verbose(false){}
+TplUnitTester::~TplUnitTester() {}
+
+void TplUnitTester::setVerbose(bool new_verbose){ verbose = new_verbose; }
+
+bool TplUnitTester::Test(UnitTests test){
+    if (test == unit_all) {
+        cout << endl << endl;
+        cout << "---------------------------------------------------------------------" << endl;
+        cout << "       Tasmanian Third Party Libraries (TPL): Unit Tests" << endl;
+        cout << "---------------------------------------------------------------------" << endl << endl;
+        bool pass = true;
+        pass = testTriToeplitzEigen();
+        cout << endl;
+        if (pass){
+            cout << "---------------------------------------------------------------------" << endl;
+            cout << "           All Unit Tests Completed Successfully" << endl;
+            cout << "---------------------------------------------------------------------" << endl << endl;
+        } else {
+            cout << "FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL" << endl;
+            cout << "         Some Unit Tests Have Failed" << endl;
+            cout << "FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL" << endl << endl;
+        }
+        return pass;
+    } else if (test == unit_none) {
+        cout << "No tests to run.\n";
+        return true;
+    } else {
+        cout << "Unknown test type!\n";
+        return false;
+    }
+}
+
+bool testTriToeplitzEigen() {
+    // Initialize.
+    const int N = 100;
+    bool is_matched;
+    double exact_eigs[N], D[N], E[N-1], W[N], Z[N*N], WORK1[2 * N - 2], WORK2[4 * N];
+    int IBLOCK1[N], ISPLIT1[N], IWORK1[3 * N];
+    double a = M_PI_2;
+    double b = M_PI_4;
+    ISPLIT1[0] = N;
+    IBLOCK1[0] = N;
+    for (int i=0; i<N; i++) {
+        exact_eigs[i] = a + 2.0 * b * cos((i+1) * M_PI / (N+1));
```

Comment:
The MS compiler doesn't like `M_PI`, use `TasGrid::Maths::pi` defined in `tsgMathUtils.hpp`.

Also, use `std::cos()`